service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/AbstractServiceClient.java

```diff
@@ -160,6 +160,18 @@ protected String encodeUrl(String url) {
     protected String getOptURI(String paramName, Object paramValue) {
         if (paramValue == null || paramValue instanceof String && paramValue.toString().isEmpty()) {
             return "";
+        } else if (paramValue instanceof List) {
+            StringBuilder builder = new StringBuilder();
+            String delimiter = "";
+            List<String> paramValueList = (List<String>) paramValue;
```

Comment:
No, the primary concern is should we do this based on type of object? 
Ex: at api layer, poseidon accepts 
"datatype": "integer",
"multivalue": true

Internally, poseidon reads all param values for the same param name and gives as List of integers to app.

So even at service client definition, its natural to say
"type": "String|long|int",
"multivalue": true

rather than
"type": "java.util.List < String|long|int >"

If we go with this approach (adding multivalue: true) we need to change generation code. This can't be done only at AbstractServiceClient